File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        found = False
        base = 0

        while not found:
            mid = nums[len(nums) // 2]
            if mid == target:
                return base + (len(nums) // 2)
            if len(nums) == 1:
                return -1
            if mid > target:
                nums = nums[:(len(nums) // 2)]
                continue
            if mid < target:
                base += len(nums) // 2
                nums = nums[(len(nums) // 2):]
                continue
                

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int length = nums.length;
        int[] newArr = new int[length * 2];

        for (int i = 0, j = length; i < length; i++, j++) {
            newArr[i] = nums[i];
            newArr[j] = nums[i];
        }

        return newArr;
    }
}